src/T2DMap.cpp

```diff
@@ -855,7 +856,7 @@ void T2DMap::paintEvent(QPaintEvent* e)
             oneWayExits.clear();
             if (!room->customLines.empty()) {
                 // This room has custom exit lines:
-                if (!room->customLines.contains(QStringLiteral("n"))) {
+                if (!room->customLines.contains(QLatin1String("n"))) {
```

Comment:
...but will it be a brand **new** `QString` object - I thought that all of the same raw string will be shared, at least within the compilation unit if not the whole executable so the creation cost is divided by the total number of uses...?

Ah, yeah, here we go (I think we have referred to [this](https://blog.qt.io/blog/2014/06/13/qt-weekly-13-qstringliteral/) before):
> ### Avoid duplicated QStringLiterals
>
> Avoid having multiple QStringLiterals with the same content. For plain literals and QLatin1String, compilers try to consolidate identical literals so that they are not duplicated. For QStringLiteral, identical strings cannot be merged.

Reply:
I think that is just not duplicated within the binary (think size). We still pay for the cost of creating every single one at runtime.

Reply:
How do you come to that conclusion? It is likely - but need not be apparently - that there will be a single copy of the `QLatin1String` in the memory taken up by the application which will be used in all places in the code which use it... Fair enough - apparently sharing literal strings (which I believe is what happens here) is implementation dependent according to the notes in https://en.cppreference.com/w/cpp/language/string_literal

And in https://woboq.com/blog/qstringliteral.html:
> Hence you can use `QLatin1String`, which is just a thin wrapper around `char *` that specify the encoding. There are overloads taking `QLatin1String` for functions that can operate or the raw latin1 data directly without conversion.

Reply:
It's just nicer to have code that's not lot up like a Christmas tree with warnings. I've read the documentation, woboq and Qt blog on this the same...

Reply:
:1st_place_medal: Okay - gone for the optimum solution that should keep ***everyone*** happy - with 12 `const QString& key_XXXX` defined at the top of the `T2DMap.cpp` file from `QStringLiteral`s and then used throughout the file wherever the raw literal was previously used...!